refactor(init): Improve error handling for `graphId` step, improve and code structure of `rover init` (#2539)
apollographql/apollo-cli: docs/source/errors.md

> 

Author: @Meschreiber We've added a new [descriptive error](https://github.com/apollographql/rover/pull/2539/files#diff-4438358ff297f814a7206893ae805f186518f4fbfff728555df0dfe25af696e7) for when users attempt to generate a GraphOS project with `rover init` but experience a failure outside of their control (i.e. network/server), which means we need to update [this page](https://www.apollographql.com/docs/rover/errors) -- let me know if this could be updated or refined 👍 !

Reviewer: Hooray for descriptive errors! I would advocate for calling the thing a "graph" rather than project since that's what the CLI calls it to the best of my knowledge. Will make some inline suggestions.

Offline enterprise license support for Apollo is available on an as-needed basis. It must be enabled on your Studio organization. For access, send a request to your Apollo contact.

### E045

> This error occurs when Rover tries to create a project but creating the project fails.

Reviewer: This error can only be thrown from `init`, right? Let's link out to the command to make that clear.

```suggestion
This error occurs when Rover tries to [create a graph](/rover/commands/init) but fails.
```

> This is likely a temporary issue and you can try again. If the problem persists, please contact [Apollo Support](https://support.apollographql.com).

Reviewer: Suggestion to separate out the "Temporary" part (most important), and then the next steps.

```suggestion
This is likely temporary. Please try again or contact [Apollo Support](https://support.apollographql.com) if the issue persists.
```
